Lib/traceback.py

```diff
@@ -295,11 +301,17 @@ def __repr__(self):
     def __len__(self):
         return 4
 
+    @property
+    def _original_line(self):
+        # Returns the line as-is from the source, without modifying whitespace.
+        self.line
```

Comment:
I'd personally prefer making `_line` a `cachedproperty` instead of this redundantly-seeming (though effective) statement

Reply:
Aah that's a great idea, I feel like it would make this a lot more readable, there's another empty `self.line` in the constructor if you pass `lookup_line`. Unfortunately I tried it and it looks like `cached_property` from functools doesn't support classes with `__slots__`:

> Also, this decorator requires that the `__dict__` attribute on each instance be a mutable mapping. This means it will not work with some types, such as metaclasses (since the `__dict__` attributes on type instances are read-only proxies for the class namespace), and those that specify `__slots__` without including `__dict__` as one of the defined slots (as such classes don’t provide a `__dict__` attribute at all).

Or did you mean `cachedproperty` from somewhere else?

Reply:
Ah, I wasn't aware of that restriction :/ Well, then let's keep it this way.